packages/consumption/test/modules/notifications/itemProcessors/attributeDeleted/ThirdPartyOwnedRelationshipAttributeDeletedByPeerNotificationItemProcessor.test.ts

```diff
@@ -0,0 +1,203 @@
+        expect(checkResult.isError()).toBe(false);
+
+        /* Run process() and validate its results. */
+        const event = await processor.process(notificationItem, notification);
+        expect(event).toBeInstanceOf(ThirdPartyOwnedRelationshipAttributeDeletedByPeerEvent);
+        const updatedAttribute = event?.data;
+        expect(notificationItem.attributeId.equals(updatedAttribute!.id)).toBe(true);
+        expect(updatedAttribute!.deletionInfo?.deletionStatus).toStrictEqual(DeletionStatus.DeletedByPeer);
+
+        /* Manually trigger and verify rollback. */
+        await processor.rollback(notificationItem, notification);
+        const attributeAfterRollback = await consumptionController.attributes.getLocalAttribute(notificationItem.attributeId);
+        expect(attributeAfterRollback?.deletionInfo).toBeUndefined();
+    });
+
+    test("runs all processor methods for an unknown attribute", async function () {
+        const unknownAttributeId = CoreId.from("id1xxxxxxxxxxxxxxxxx");
+
+        const notificationItem = ThirdPartyOwnedRelationshipAttributeDeletedByPeerNotificationItem.from({
+            attributeId: unknownAttributeId
+        });
+        const notification = LocalNotification.from({
+            id: CoreId.from("notificationRef"),
+            source: LocalNotificationSource.from({
+                type: "Message",
+                reference: CoreId.from("messageRef")
+            }),
+            status: LocalNotificationStatus.Open,
+            isOwn: false,
+            peer: CoreAddress.from("peer"),
+            createdAt: CoreDate.utc(),
+            content: Notification.from({
+                id: CoreId.from("notificationRef"),
+                items: [notificationItem]
+            }),
+            receivedByDevice: CoreId.from("deviceId")
+        });
+        const processor = new ThirdPartyOwnedRelationshipAttributeDeletedByPeerNotificationItemProcessor(consumptionController);
+
+        /* Run and check validation. */
+        const checkResult = await processor.checkPrerequisitesOfIncomingNotificationItem(notificationItem, notification);
+        expect(checkResult.isError()).toBe(false);
+
+        /* Run process() and validate its results. */
+        const processResult = await processor.process(notificationItem, notification);
+        expect(processResult).toBeUndefined();
+
+        /* Manually trigger and verify rollback. */
+        const rollbackResult = await processor.rollback(notificationItem, notification);
+        expect(rollbackResult).toBeUndefined();
+    });
+
+    test("detects spoofing attempts", async function () {
+        /* A naughty peer is trying to delete attributes shared
+         * not with him, but with another peer. This must be
+         * caught by the validation. */
+        const differentThirdPartyOwnedRelationshipAttribute = await consumptionController.attributes.createAttributeUnsafe({
```

Comment:
IMO the word `different` can be omitted when naming the variable `differentThirdPartyOwnedRelationshipAttribute.`, as only one ThirdPartyOwnedRelationshipAttribute plays a role in this test.

Reply:
The variable name is chosen to stress the fact, that it is a ThirdPartyRelationshipAttribute in the context of a different Relationship (as you have to look thoroughly for the difference and we don't use comments).

Reply:
All right, maybe you can think of an adjective to emphasize this more, because it wasn't quite clear to me, but if not it is good like this 👍